models/intermediate/int_institutional_claim_deduped.sql

```diff
@@ -0,0 +1,844 @@
+        , atndg_prvdr_npi_num
+        , clm_adjsmt_type_cd
+        , clm_efctv_dt
+        , clm_admsn_type_cd
+        , clm_admsn_src_cd
+        , clm_bill_freq_cd
+        , dgns_prcdr_icd_ind
+        , clm_mdcr_instnl_tot_chrg_amt
+        , clm_blg_prvdr_oscar_num
+        , file_name
+        , file_date
+        , row_num
+    from {{ ref('int_institutional_claim_adr') }}
+
+)
+
+, diagnosis_pivot as (
+
+    select * from {{ ref('int_diagnosis_pivot') }}
+
+)
+
+, procedure_pivot as (
+
+    select * from {{ ref('int_procedure_pivot') }}
+
+)
+
+, revenue_center as (
+
+    select * from {{ ref('int_revenue_center_deduped') }}
+
+)
+
+/*
+    sum the adjusted header amounts
+
+    (CCLF docs ref: 5.3 Calculating Beneficiary-Level Expenditures)
+*/
+, header_totals as (
+
+    select
+          clm_blg_prvdr_oscar_num
+        , clm_from_dt
+        , clm_thru_dt
+        , current_bene_mbi_id
+        , sum(clm_pmt_amt) as sum_clm_pmt_amt
+        , sum(clm_mdcr_instnl_tot_chrg_amt) as sum_clm_mdcr_instnl_tot_chrg_amt
+    from sort_adjusted_claims
+    group by
+          clm_blg_prvdr_oscar_num
+        , clm_from_dt
+        , clm_thru_dt
+        , current_bene_mbi_id
+
+)
+
+/*
+    apply final adjustment logic by selecting latest version of claim,
+    removing any remaining claims with a canceled status, and adding header totals.
+*/
+, filter_claims as (
+
+    select
+          sort_adjusted_claims.cur_clm_uniq_id
+        , sort_adjusted_claims.current_bene_mbi_id
+        , sort_adjusted_claims.clm_from_dt
+        , sort_adjusted_claims.clm_thru_dt
+        , sort_adjusted_claims.clm_bill_fac_type_cd
+        , sort_adjusted_claims.clm_bill_clsfctn_cd
+        , header_totals.sum_clm_pmt_amt as clm_pmt_amt
+        , sort_adjusted_claims.bene_ptnt_stus_cd
+        , sort_adjusted_claims.dgns_drg_cd
+        , sort_adjusted_claims.fac_prvdr_npi_num
+        , sort_adjusted_claims.atndg_prvdr_npi_num
+        , sort_adjusted_claims.clm_adjsmt_type_cd
+        , sort_adjusted_claims.clm_efctv_dt
+        , sort_adjusted_claims.clm_admsn_type_cd
+        , sort_adjusted_claims.clm_admsn_src_cd
+        , sort_adjusted_claims.clm_bill_freq_cd
+        , sort_adjusted_claims.dgns_prcdr_icd_ind
+        , header_totals.sum_clm_mdcr_instnl_tot_chrg_amt as clm_mdcr_instnl_tot_chrg_amt
+        , sort_adjusted_claims.clm_blg_prvdr_oscar_num
+        , sort_adjusted_claims.file_name
+        , sort_adjusted_claims.file_date
+    from sort_adjusted_claims
+        left join header_totals
+            on sort_adjusted_claims.clm_blg_prvdr_oscar_num = header_totals.clm_blg_prvdr_oscar_num
+            and sort_adjusted_claims.clm_from_dt = header_totals.clm_from_dt
+            and sort_adjusted_claims.clm_thru_dt = header_totals.clm_thru_dt
+            and sort_adjusted_claims.current_bene_mbi_id = header_totals.current_bene_mbi_id
+    where sort_adjusted_claims.row_num = 1
+    and sort_adjusted_claims.clm_adjsmt_type_cd <> '1'
```

Comment:
Based on my exploration of real data, CMS does not always send the proper adjustment type codes for adjusted claims. They were often sending a '0' instead of a '2' for an adjusted claim, so it was unreliable in the sorting logic. This is why we are using the latest claim effective date and claim ID to try to ensure the adjusted claims get selected rather than the canceled claims. Again, this is all based on my own testing with real data (from multiple claim years), but everyone's dataset may be different.

Do you have a suggestion for better sorting logic based on your experience?